.github/workflows/build-for-release.yml

```diff
@@ -78,30 +78,28 @@ jobs:
       uses: actions/checkout@v3
       with:
         fetch-depth: 0  # IMPORTANT! to get all the branches
-    - name: Ensure that the specified commit exists in the latest release branch
-      env:
-        COMMIT: ${{ inputs.commit }}
+        sparse-checkout: release
+    - name: Prepare build scripts
+      run: cd ${{ github.workspace }}/release && yarn && yarn build
+    - name: Get Release Branch
+      uses: actions/github-script@v6
+      id: release_branch
+      with:
+        result_encoding: string
+        script: | # js
+          const { getReleaseBranch } = require('${{ github.workspace }}/release/dist/index.cjs');
+          const version = '${{ inputs.version }}';
+          const releaseBranch = getReleaseBranch(version);
+          return releaseBranch;
+    - name: Ensure that the specified commit exists in the ${{ steps.release_branch.outputs.result }} release branch
       run: |
-        echo "Checking if the specified commit is in a release branch..."
-
-        git branch -a --contains $COMMIT > branches.txt
-        if [[ $(grep -c master branches.txt) =~ 1 ]]; then
-          echo "Found in master branch. ABORT!"
-          exit -1
-        fi
-        if [[ $(grep -c 'release-x' branches.txt) =~ 1 ]]; then
-          echo "Found the commit $COMMIT in:"
-          git branch -a --contains $COMMIT
-          echo "Proceeding to the next step..."
-          exit 0
-        else
-          echo "Commit $COMMIT is not found in a single release branch"
-          echo "ABORT!."
-          exit -1
-        fi
+        git checkout ${{ steps.release_branch.outputs.result }}
+        git branch --contains ${{ inputs.commit }} | grep -q ${{ steps.release_branch.outputs.result }} \
+          && echo "Commit found in correct release branch" \
+          || (echo "Commit not found in correct release branch" && exit 1)
```

Comment:
Nit:

I know you find this pattern weird, but you can have `${{ steps.release_branch.outputs.result }}` stored as `env` for this `run`.

```yml
run: echo $BRANCH
env:
  BRANCH: ${{ steps.release_branch.outputs.result }}
```

Reply:
that makes sense here, I just prefer the `env` before the `run`. will update

Reply:
interesting. using `env` added extra quotes and cause failures. but if I set it in bash it works just fine.
![Screen Shot 2023-10-12 at 7 47 37 AM](https://github.com/metabase/metabase/assets/30528226/a75b7bdf-2047-4353-b046-8dcc693816b3)